packages/teams-js/src/private/interactive.ts

```diff
@@ -0,0 +1,207 @@
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+
+      resolve(sendAndHandleSdkError('interactive.getFluidContainerId'));
+    });
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   * ------
+   * Sets the ID of the fluid container associated with the current context.
+   *
+   * @remarks
+   * If this returns false, the client should delete the container they created and then call
+   * `getFluidContainerId()` to get the ID of the container being used.
+   * @param containerId ID of the fluid container the client created.
+   * @returns True if the client created the container that's being used.
+   */
+  export function setFluidContainerId(containerId: string): Promise<boolean> {
+    return new Promise<boolean>(resolve => {
+      ensureInitialized(FrameContexts.meetingStage, FrameContexts.sidePanel);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+
+      resolve(sendAndHandleSdkError('interactive.setFluidContainerId', containerId));
+    });
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   * ------
+   * Returns the shared clock server's current time.
+   */
+  export function getNtpTime(): Promise<NtpTimeInfo> {
+    return new Promise<NtpTimeInfo>(resolve => {
+      ensureInitialized(FrameContexts.meetingStage, FrameContexts.sidePanel);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+
+      resolve(sendAndHandleSdkError('interactive.getNtpTime'));
+    });
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   * ------
+   * Associates the fluid client ID with a set of user roles.
+   *
+   * @param clientId The ID for the current user's Fluid client. Changes on reconnects.
+   * @returns The roles for the current user.
+   */
+  export function registerClientId(clientId: string): Promise<UserMeetingRole[]> {
+    return new Promise<UserMeetingRole[]>(resolve => {
+      ensureInitialized(FrameContexts.meetingStage, FrameContexts.sidePanel);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+
+      resolve(sendAndHandleSdkError('interactive.registerClientId', clientId));
+    });
+  }
+
+  /**
+   * @hidden
+   * Hide from docs
+   * ------
+   * Returns the roles associated with a client ID.
+   *
+   * @param clientId The Client ID the message was received from.
+   * @returns The roles for a given client. Returns `undefined` if the client ID hasn't been registered yet.
+   */
+  export function getClientRoles(clientId: string): Promise<UserMeetingRole[] | undefined> {
+    return new Promise<UserMeetingRole[] | undefined>(resolve => {
+      ensureInitialized(FrameContexts.meetingStage, FrameContexts.sidePanel);
+      if (!isSupported()) {
+        throw errorNotSupportedOnPlatform;
+      }
+
+      resolve(sendAndHandleSdkError('interactive.getClientRoles', clientId));
+    });
+  }
+
+  export function isSupported(): boolean {
```

Comment:
See my other comment. I think for now it would be best to not include this in the capability model and keep it separate, without isSupported checks. Especially if it's not meant to be called by developers directly and if it's supposed to work in Teams (both angular and tmp) before they integrate the host sdk.

Reply:
If you do want to be part of the capability model, this will need to be a new capability, and not part of `meeting`, as there are (or might be in the future) clients who will want support `meeting` but will not support this `interactive` stuff.